src/command/repo/get.c

```diff
@@ -46,13 +49,81 @@ storageGetProcess(IoWrite *destination)
+
+                        // If stanza option is specified, it must match the given file path
+                        if (cfgOptionStrNull(cfgOptStanza) != NULL && ! strEq(stanza, cfgOptionStr(cfgOptStanza)))
+                            THROW_FMT(AssertError, "stanza name '%s' given in option doesn't match the given path", 
+                                strPtr(cfgOptionStr(cfgOptStanza)));
+
+                        if (strEq(strLstGet(filePathSplitLst, 0), STORAGE_PATH_ARCHIVE_STR))
+                        {
+                            // Find the archiveCipherPass
+                            if (! strEndsWithZ(relativeFile, INFO_ARCHIVE_FILE) &&
+                                ! strEndsWithZ(relativeFile, INFO_ARCHIVE_FILE".copy"))
+                            {
+                                InfoArchive *info = infoArchiveLoadFile(
+                                    storageRepo(), strNewFmt(STORAGE_PATH_ARCHIVE "/%s/%s", strPtr(stanza), INFO_ARCHIVE_FILE),
+                                    repoCipherType, cipherPass);
+                                cipherPass = strDup(infoArchiveCipherPass(info));
+                            }
+                            
+                        }
+                        
+                        if (strEq(strLstGet(filePathSplitLst, 0), STORAGE_PATH_BACKUP_STR))
+                        {
+                            if (! strEndsWithZ(relativeFile, INFO_BACKUP_FILE) &&
+                                ! strEndsWithZ(relativeFile, INFO_BACKUP_FILE".copy"))
+                            {
+                                // Find the backupCipherPass
+                                InfoBackup *info = infoBackupLoadFile(
+                                    storageRepo(), strNewFmt(STORAGE_PATH_BACKUP "/%s/%s", strPtr(stanza), INFO_BACKUP_FILE),
+                                    repoCipherType, cipherPass);
+                                cipherPass = strDup(infoBackupCipherPass(info));
+
+                                // Find the manifestCipherPass
+                                if (! strEq(strLstGet(filePathSplitLst, 2), STRDEF("backup.history")) &&
+                                    ! strEndsWithZ(relativeFile, BACKUP_MANIFEST_FILE) &&
+                                    ! strEndsWithZ(relativeFile, BACKUP_MANIFEST_FILE".copy"))
+                                {
+                                   const Manifest *manifest = manifestLoadFile(
+                                        storageRepo(), strNewFmt(STORAGE_PATH_BACKUP "/%s/%s/%s", strPtr(stanza), 
+                                        strPtr(strLstGet(filePathSplitLst, 2)), BACKUP_MANIFEST_FILE), repoCipherType, cipherPass);
+                                    cipherPass = strDup(manifestCipherSubPass(manifest));
+                                }
+                            }
+                        }
+                    }
+                }
+
```

Comment:
I think there needs to be some sort of error here. Currently no files can be above the archive/backup directories so it's not clear what key would be used if one did exist.

Reply:
With the current code, `cfgOptRepoCipherPass` will be the default if no `cfgOptCipherPass` is given.
That match the repo-put possibilities also (and the previous tests written).

It would be easy to modify to:
```
if (archive)..
else if (backup)..
else AssertError
```

But would it really be useful since we can use the default RepoCipherPass if no specific CipherPass is provided?
In the future, if something is added, we can still modify this. I just assumed that since archive/backup.info are encrypted using the RepoCipherPass, if we add an extra level, it would use the same idea.

Reply:
It's hard to say what we might do in the future so I'd rather not try to plan ahead. What I'm worried about is if the user puts some file/dir at the top level of the repo and we try to decrypt it with the master key. That's not likely to work and there should not be any files there anyway. We should be throwing an error, something like `unable to determine encryption key for '/path'`. The error type can be FormatError.

Reply:
Ok.

Added the else if as mentioned above and:

```
else
{
    // Nothing should be stored at the top level of the repo
    THROW_FMT(FormatError, "unable to determine encryption key for '%s'", strPtr(relativeFile));
}
```

(I'll have to modify the tests once you think the get.c code is ready ^^)

Reply:
Looks good, just use `OptionInvalidValueError` instead of `FormatError`.

Reply:
Done